src/System.Linq.Expressions/src/System/Dynamic/Utils/TypeExtensions.cs

```diff
@@ -68,9 +68,7 @@ internal static ParameterInfo[] GetParametersCached(this MethodBase method)
             if (!pic.TryGetValue(method, out ParameterInfo[] pis))
             {
                 pis = method.GetParameters();
-
-                Type t = method.DeclaringType;
-                if (t != null && t.CanCache())
+                if (method.DeclaringType?.IsCollectible == false)
```

Comment:
@jkotas this is the case I mentioned where really what we want is a `MethodInfo.IsCollectible`, though continuing to assume that a method without a declaring type isn't collectible is still and improvement in the meantime.

Reply:
I think having IsCollectible at each level make sense. Update proposal in #25671 to expose it on MemberInfo (that covers all types, methods and fields) and Assembly?